agent/consul/intention_endpoint_test.go

```diff
@@ -273,6 +273,44 @@ func TestIntentionApply_updateGood(t *testing.T) {
 	}
 }
 
+// TestIntentionApply_NoSourcePeer makes sure that no intention is created with a SourcePeer since this is not supported
+func TestIntentionApply_NoSourcePeer(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+
+	dir1, s1 := testServer(t)
+	defer os.RemoveAll(dir1)
+	defer s1.Shutdown()
```

Comment:
these defers shouldn't be necessary (unsure about the codec)

Reply:
codec `Close()` can also be removed since it's in a `t.Cleanup()` of `rpcClient`

Reply:
🤔 the set up was gentle copy pasta 🍝 from other tests; let me have a look and happy to break the mold here.

Reply:
So `os.RemoveAll(dir)` is not actually called in the cleanup; perhaps we can do a big refactor later per this but that can come in a separate PR

Reply:
> So os.RemoveAll(dir) is not actually called in the cleanup; perhaps we can do a big refactor later per this but that can come in a separate PR

🤦🏼 ah, I didn't actually look at `testutil.TempDir` :( -- thanks for calling that out Chris !! 🥇